tests/virtual_machines_test.go

```diff
@@ -389,6 +390,117 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
+					BeforeEach(func() {
+						for numOfIfaces := 1; numOfIfaces <= maxNumOfIfaces; numOfIfaces++ {
+							By(fmt.Sprintf("updating the number of interfaces to %d", numOfIfaces))
+							err := retryOnConflictAndWaitForRestoreCache(vm, extendedTimeout, func(vm *kubevirtv1.VirtualMachine) {
+								vm.Spec.Template.Spec.Domain.Devices.Interfaces = interfaces[0:numOfIfaces]
+								vm.Spec.Template.Spec.Networks = networks[0:numOfIfaces]
+							})
+							Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Should succeed updating the vm to %d interfaces", numOfIfaces))
+						}
+
+						By(fmt.Sprintf("Checking that the vm has %d interface after the changes are made", maxNumOfIfaces))
+						err := testClient.VirtClient.Get(context.TODO(), client.ObjectKey{Namespace: vm.Namespace, Name: vm.Name}, vm)
+						Expect(err).ToNot(HaveOccurred(), "Should succeed getting vm")
+						Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces).To(HaveLen(maxNumOfIfaces), fmt.Sprintf("Should have %d interfaces", maxNumOfIfaces))
+					})
+					It("should not be able to allocate the occupied macs to new vms", func() {
+						By(fmt.Sprintf("Creating %d vms, each with a MAC interface used in the base vm", maxNumOfIfaces))
+						for ifaceIdx := 0; ifaceIdx < maxNumOfIfaces; ifaceIdx++ {
+							newVm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{interfaces[ifaceIdx]}, []kubevirtv1.Network{networks[ifaceIdx]})
+							err := testClient.VirtClient.Create(context.TODO(), newVm)
+							Expect(err).To(HaveOccurred(), "Should fail creating the vm")
+						}
+					})
+				})
+			})
+			Context("and a VM's NIC is added just before an old update request with no NICs is made", func() {
+				var vm *kubevirtv1.VirtualMachine
+				var reusedMacAddress string
+				BeforeEach(func() {
+					By("Creating a vm with no NICs")
+					vm = CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{}, []kubevirtv1.Network{})
+					err := testClient.VirtClient.Create(context.TODO(), vm)
+					Expect(err).ToNot(HaveOccurred(), "Should succeed creating the vm")
+
+					By("Saving the vm instance to reuse it after resourceVersion has changed to cause a conflict error")
+					copyVm := vm.DeepCopy()
+
+					By("Adding a new NIC to the vm")
+					err = retryOnConflictAndWaitForRestoreCache(vm, extendedTimeout, func(vm *kubevirtv1.VirtualMachine) {
+						vm.Spec.Template.Spec.Domain.Devices.Interfaces = []kubevirtv1.Interface{newInterface("br1", "")}
+						vm.Spec.Template.Spec.Networks = []kubevirtv1.Network{newNetwork("br1")}
+					})
+					Expect(err).ToNot(HaveOccurred(), "Should succeed updating the vm")
+					reusedMacAddress = vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+					_, err = net.ParseMAC(reusedMacAddress)
+					Expect(err).ToNot(HaveOccurred(), "Should succeed parsing the vm mac")
+
+					By("Issuing the outdated vm request")
```

Comment:
there is already a polarion test (test_id: 2995) doing that.

Reply:
I don't think so. it would basically be us checking k8s conflict mechanism
This test expects the request to fail, if this ever breaks we'd be able to follow the `By` steps and understand it with out issuing another test